src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -36,16 +33,20 @@ internal class ExternalErrorDiagnosticUpdateSource : IDiagnosticUpdateSource
         private InprogressState _stateDoNotAccessDirectly = null;
         private ImmutableArray<DiagnosticData> _lastBuiltResult = ImmutableArray<DiagnosticData>.Empty;
 
-        [ImportingConstructor]
         public ExternalErrorDiagnosticUpdateSource(
             VisualStudioWorkspaceImpl workspace,
             IDiagnosticAnalyzerService diagnosticService,
             IDiagnosticUpdateSourceRegistrationService registrationService,
-            IAsynchronousOperationListenerProvider listenerProvider) :
+            IAsynchronousOperationListenerProvider listenerProvider,
+            IKnownUIContextService uiContextService) :
                 this(workspace, diagnosticService, registrationService, listenerProvider.GetListener(FeatureAttribute.ErrorList))
         {
-            Debug.Assert(!KnownUIContexts.SolutionBuildingContext.IsActive);
-            KnownUIContexts.SolutionBuildingContext.UIContextChanged += OnSolutionBuild;
+            // it can be null in unit test
+            if (uiContextService != null)
+            {
+                Debug.Assert(!uiContextService.IsSolutionBuilding);
```

Comment:
This assert seems questionable -- Roslyn could get loaded while build (say of C++ only projects) is running.